spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/reload/KubernetesClientEventBasedConfigMapChangeDetector.java

```diff
@@ -111,34 +110,42 @@ void inform() {
 
 		namespaces.forEach(namespace -> {
 			SharedIndexInformer<V1ConfigMap> informer;
-			String filter = null;
+			String[] filter = new String[1];
 
 			if (enableReloadFiltering) {
-				filter = ConfigReloadProperties.RELOAD_LABEL_FILTER + "=true";
-				LOG.debug(() -> "added configmap informer for namespace : " + namespace + " with enabled filter");
-			}
-			else {
-				LOG.debug(() -> "added configmap informer for namespace : " + namespace);
+				filter[0] = ConfigReloadProperties.RELOAD_LABEL_FILTER + "=true";
 			}
 
-			String filterOnInformerLabel = filter;
-			informer = factory
-					.sharedIndexInformerFor(
-							(CallGeneratorParams params) -> coreV1Api.listNamespacedConfigMapCall(namespace, null, null,
-									null, null, filterOnInformerLabel, null, params.resourceVersion, null,
-									params.timeoutSeconds, params.watch, null),
-							V1ConfigMap.class, V1ConfigMapList.class);
+			// We need to pass an APIClient to the SharedInformerFactory because if we use
+			// the
+			// default
+			// constructor it will use the configured default APIClient but that may not
+			// contain
+			// an APIClient configured within the cluster and does not contain the
+			// necessary
+			// certificate authorities for the cluster. This results in SSL errors.
+			// See https://github.com/spring-cloud/spring-cloud-kubernetes/issues/885
+			SharedInformerFactory factory = new SharedInformerFactory(apiClient);
```

Comment:
the problem is basically on this line. What I used to do is take a `SharedInformerFactory` and try to create many informers for different namespaces via multiple calls to `sharedIndexInformerFor`. Again: I was doing that on the same instance of `SharedInformerFactory`. Unfortunately, a `SharedInformerFactory` does not work like that (+ I am a total idiot for trusting my understanding and not writing a proper integration test).

A `SharedInformerFactory` has a cache internally on the Type of the informer and its instance. Since in my case I had the same type (`V1ConfigMap` for example), this was never working. I was never basically adding a new informer, all the actions happened on the same one. What was happening is that on the _same_ (I can not stress this enough) instance of an informer, I had two handlers. 

The fix after the above is quite trivial, create a new factory for each namespace and a new informer. Close informers and factories when done.

Now, what could happen if there are many namespaces? Potentially, these are calls to the API Server and I doubt that many of them will make the api-server happy, but I'll leave this one open in my mind and may be ask around on the kubernetes-client project. I hope this is OK for now.

Reply:
Basically all the informers for each namespace will be making a lot of API calls?

Where as if we could use a single informer they could make one API call for all configmap data for each namespace?

Reply:
now that you made your comment, I realize that the same number of calls would be made; even if we could have used a single informer. 

A single informer, multiple namespaces (if that would be possible): 
- since RBAC rules need to be enforced, each of those namespaces would result in a different call to the API Server

Multiple informers, one for a single namespace:

- the same number of calls as above.

Reply:
build is fine now, can be looked at again. thank you

Reply:
That makes sense